src/command_api.rs

```diff
@@ -29,6 +30,118 @@ pub struct JuliaupApiGetinfoReturn {
 }
 
+fn get_channel_info(
+    name: &String,
+    channel: &JuliaupConfigChannel,
+    config_file: &JuliaupReadonlyConfigFile,
+    paths: &GlobalPaths,
+) -> Result<Option<JuliaupChannelInfo>> {
+    match channel {
+        JuliaupConfigChannel::SystemChannel {
+            version: fullversion,
+        } => {
+            let (platform, mut version) = parse_versionstring(fullversion)
+                .with_context(|| "Encountered invalid version string in the configuration file while running the getconfig1 API command.")?;
+
+            version.build = semver::BuildMetadata::EMPTY;
+
+            match config_file.data.installed_versions.get(&fullversion.clone()) {
+                Some(channel) => Ok(Some(JuliaupChannelInfo {
+                    name: name.clone(),
+                    file: paths.juliauphome
+                        .join(&channel.path)
+                        .join("bin")
+                        .join(format!("julia{}", std::env::consts::EXE_SUFFIX))
+                        .normalize()
+                        .with_context(|| "Normalizing the path for an entry from the config file failed while running the getconfig1 API command.")?
+                        .into_path_buf()
+                        .to_string_lossy()
+                        .to_string(),
+                    args: Vec::new(),
+                    version: version.to_string(),
+                    arch: platform
+                })),
+                None => bail!("The channel '{}' is configured as a system channel, but no such channel exists in the versions database.", name)
+            }
+        }
+        JuliaupConfigChannel::LinkedChannel { command, args } => {
+            let mut new_args: Vec<String> = Vec::new();
+
+            for i in args.as_ref().unwrap() {
+                new_args.push(i.to_string());
+            }
+
+            new_args.push("--version".to_string());
+
+            let res = std::process::Command::new(command)
+                .args(&new_args)
+                .output();
+
+            match res {
+                Ok(output) => {
+                    let expected_version_prefix = "julia version ";
+
+                    let trimmed_string = std::str::from_utf8(&output.stdout).unwrap().trim();
+
+                    if !trimmed_string.starts_with(expected_version_prefix) {
+                        return Ok(None);
+                    }
+
+                    let version =
+                        Version::parse(&trimmed_string[expected_version_prefix.len()..])?;
+
+                    Ok(Some(JuliaupChannelInfo {
+                        name: name.clone(),
+                        file: command.clone(),
+                        args: args.clone().unwrap_or_default(),
+                        version: version.to_string(),
+                        arch: "".to_string(),
+                    }))
+                }
+                Err(_) => Ok(None),
+            }
+        }
+        // TODO: fix
```

Comment:
@christiangnrd I missed this TODO before. What does it refer to? Can you elaborate

Reply:
Fair enough! Sorry this PR lingered for so long!